### DISCUSSION TITLE
What are some security measures considered to misuse external connectivity of resources via `skypilot exec/ ssh`?

### DISCUSSION THREAD

**Discussion by aavarghese** [skypilot-org/skypilot#3654] [Q&A]
We have used Skypilot to launch some example tasks on IBM Cloud VPC based instances. Since skypilot exposes SSH access to clusters via floating IPs, this brings up few concerns regarding security and scalability of such resources. Today it is **required** to have an external IP created to support some of the skypilot commands post provisioning and setup, correct? How can we guarantee our instances are secure?

**Comment by romilbhardwaj** [comment]
Hi @aavarghese - SkyPilot supports private IPs on provisioned instances by running them in a VPC. This is currently supported on [AWS](https://skypilot.readthedocs.io/en/latest/cloud-setup/cloud-permissions/aws.html#using-a-specific-vpc) and [GCP](https://skypilot.readthedocs.io/en/latest/cloud-setup/cloud-permissions/gcp.html#using-a-specific-vpc). 

Private IPs on IBM cloud is not yet supported, but we would be happy to welcome community contributions for it. Would you be interested in adding VPC and private IP support for IBM?
